lib/oli/delivery/paywall.ex

```diff
@@ -211,7 +211,8 @@ defmodule Oli.Delivery.Paywall do
   ) do
     discounts =
       from(d in Discount,
-        where: d.institution_id == ^institution_id and (is_nil(d.section_id)) or (d.section_id == ^id),
+        where: (d.institution_id == ^institution_id and is_nil(d.section_id))  # Institution-wide discounts
+          or (d.section_id == ^id and d.institution_id == ^institution_id), # Section-specific discounts for the given institution
```

Comment:
This works, but I think the bug is happening because of the lack of parentheses:

```elixir
where: d.institution_id == ^institution_id and (is_nil(d.section_id) or d.section_id == ^id),
```

Reply:
Exactly! Kept the long form for readability, double-checked and if we keep the comments I believe this is more idiomatic. Updating.